sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDfsTransformer.scala

```diff
@@ -19,27 +19,108 @@
+
+    // prepare parameters
+    val transformParameters = getCustomTransformMethodParameterInfo.get
+    val mappedParameters = transformParameters.map {
+      case dfParam if dfParam.tpe <:< typeOf[DataFrame] =>
+        val paramName = dfParam.name
+        val dfName = paramName.stripPrefix("df")
+        val df = tolerantGet(dfs, dfName)
+          .getOrElse(throw NotFoundError(s"No DataFrame found with name $dfName for parameter $paramName"))
+        (dfParam, df)
+      case dsParam if dsParam.tpe <:< typeOf[Dataset[_]] =>
+        val paramName = dsParam.name
+        val dsName = paramName.stripPrefix("ds")
+        val dsType = dsParam.tpe.typeArgs.head
+        val df = tolerantGet(dfs, dsName)
+          .getOrElse(throw NotFoundError(s"No DataFrame found with name $dsName for parameter $paramName"))
+        val dfWithSelect = {
+          val columnNames = ProductUtil.classAccessorNames(dsType)
+          df.select(columnNames.map(col): _*)
+        }
+        val ds = ProductUtil.createDataset(dfWithSelect, dsType)
+        (dsParam, ds)
+      case sessionParam if sessionParam.tpe =:= typeOf[SparkSession] => (sessionParam, session)
+      case optionsParam if optionsParam.tpe =:= typeOf[Map[String, String]] => (optionsParam, options)
+      case optionalParam if optionalParam.tpe <:< typeOf[Option[_]] =>
+        val optionVal = try {
+          Some(extractOptionVal(options, optionalParam, getConverterFor(optionalParam.tpe.typeArgs.head)))
+        } catch {
+          case _: NotFoundError => optionalParam.defaultValue.map(_.asInstanceOf[Option[Any]]).getOrElse(None)
+        }
+        (optionalParam, optionVal)
+      case defaultParam if defaultParam.defaultValue.isDefined =>
+        val defaultVal = try {
+          extractOptionVal(options, defaultParam, getConverterFor(defaultParam.tpe))
+        } catch {
+          case _: NotFoundError => defaultParam.defaultValue.get
+        }
+        (defaultParam, defaultVal)
+      case otherParam => (otherParam, extractOptionVal(options, otherParam, getConverterFor(otherParam.tpe)))
+    }
+
+    // call transform method
+    val transformResult = try {
+      CustomCodeUtil.callMethod[Any](this, transformMethod, mappedParameters.map(_._2))
+    } catch {
+      case e: InvocationTargetException =>
+        // Simplify nested exception to hide reflection complexity in exceptions from custom transformer code.
+        val targetException = e.getTargetException
+        targetException.setStackTrace(e.getTargetException.getStackTrace ++ e.getStackTrace)
+        throw targetException
+    }
+
+    // distinguish between returning single and multiple DataFrames.
+    if (transformMethod.returnType =:= typeOf[Map[String, DataFrame]]) {
+      transformResult.asInstanceOf[Map[String, DataFrame]]
+    } else if (transformMethod.returnType <:< typeOf[Map[String, Dataset[_]]]) {
+      transformResult.asInstanceOf[Map[String, Dataset[_]]].mapValues(_.toDF)
```

Comment:
Well, the DataFrame (Dataset[Row]) case is handled above. 
With the typeOf[Map[String, Dataset[_]]] we then check for return values of a Dataset[XYZ]. Like that the transform function can return a Dataset[XYZ] instead of DataFrame, and CustomDfsTransformer converts it to a DataFrame for further use by SDLB.
I agree that it would be easy to use .toDF in the transform function, but using a Dataset[XYZ] as return type is a more precise definition of the return value. This can be interesting to show in the UI documentation of the transformer.